### ISSUE TITLE
Session attribute not persisting which is setting at HttpSessionListener using Spring Session Data Redis

### ISSUE THREAD

**Issue by surya0420** [spring-projects/spring-session#1788]
**Session attributes are missing/not able to fetch which are setting at HttpSessionListener**

I want to set some attributes to session when session is created using  HttpSessionListener, and I want to validate that attribute in the next level.
 

**To Reproduce**
After installing redis just install on localhost and default port, build and run the project shared on github.

**Expected behavior**
What ever attributes I am setting on session creation HttpSessionListener 
```
 public void sessionCreated(HttpSessionEvent httpSessionEvent){
        HttpSession httpSession = httpSessionEvent.getSession();
       System.out.println("inside listner class ====");
        httpSession.setAttribute("testKey","testValue");

       System.out.println("test value inside listner===>"+httpSession.getAttribute("testKey"));
    }
```
here I am able to fetch the session atrribute testKey, but after this when flow goes to servlet, I am not getting the attribute
**Sample**

I have attached the code in this[ link](https://github.com/surya0420/SpringSession/tree/main/spring-session-xml)

**Comment by chrempl** [comment]
I have the same issue. Are there any updates?

**Comment by surya0420** [comment]
No. But I have updated some work around in stackoverflow, please check and
let me know.

**Comment by chrempl** [comment]
Do you have any link?

**Comment by marcusdacoregio** [comment]
Hi everyone, what are you trying to achieve with this? Maybe if I can understand the use case better I can provide some insights.

As far as I can remember, the `HttpSessionListener` receives events outside of the filter chain, having no Spring Session filter to save the session after you performed the changes.

**Comment by chrempl** [comment]
Put some initial attributes in every created session. I'm trying to migrate an application to Spring Session and in this application, every session is initialized with an attribute in a HttpSessionListener.

**Comment by surya0420** [comment]
https://stackoverflow.com/questions/66026796/session-atributes-missing-from-spring-session-when-setting-inside-httpsessionlis/66253899#66253899


Please check this link
